tests/jobs/test_base_job.py

```diff
@@ -31,113 +31,115 @@
 from tests.test_utils.config import conf_vars
 
 
-class TestBaseJob(unittest.TestCase):
-    class TestJob(BaseJob):
-        __mapper_args__ = {
-            'polymorphic_identity': 'TestJob'
-        }
+class TestJob(BaseJob):
+    __mapper_args__ = {
+        'polymorphic_identity': 'TestJob'
+    }
 
-        def __init__(self, func, **kwargs):
-            self.func = func
-            super().__init__(**kwargs)
+    def __init__(self, func, **kwargs):
+        self.func = func
+        super().__init__(**kwargs)
 
-        def _execute(self):
-            return self.func()
+    def _execute(self):
+        return self.func()
+
+
+def test_state_success():
```

Comment:
I think during a discussion about going the "pytest-way" we agreed that we don't want to have "class-less" tests:
https://lists.apache.org/thread.html/r8896ad5bc9f2ed4bbfea7ab2be429487335d21109c25d8c32e23448e%40%3Cdev.airflow.apache.org%3E

Reply:
Hmmm, so we did, though not be any strong means, and not being able to use pytest fixtures as arguments is perhaps a valid reason to do it?

Do you think I should post it to the ML and say "if no one objects in a week I will take it as consent and merge it"?

Reply:
I think you can still use pytest fixtures with classes

Reply:
I would suggest:
```python
class MockBaseJob(BaseJob):
    ...

class TestBaseJob:
    def test_xyx(self):
         ...
```

Reply:
I just had the same thought, and yes that would probably work. It's just with subclasses of unittest.TestCase that we can't accept fixture parameters.

Reply:
(smaller diff too)

Reply:
Yes, `unittest.TestCase` is the problem